packages/codecs-data-structures/src/data-enum.ts

```diff
@@ -142,10 +127,11 @@ export function getDataEnumEncoder<TFrom extends DataEnum>(
  * @param variants - The variant decoders of the data enum.
  * @param config - A set of config for the decoder.
  */
-export function getDataEnumDecoder<T extends DataEnum>(
-    variants: DataEnumToDecoderTuple<T>,
+export function getDataEnumDecoder<const TVariants extends Variants<Decoder<any>>>(
```

Comment:
@steveluscher @buffalojoec Just wanted to point out that "Const Type Parameters" is a feature that was introduced in TypeScript 5.0. So that change officially says no to TypeScript 4 and below users.

I know we talked before about the fact that we're only planning to support TS5 but just wanted to make that explicit.

Reply:
Oooh. More TypeScript docs for me to read. My favourite.

Reply:
Uh, would this in any way have helped us to trim down the number of `getBlock()` overloads, @buffalojoec?

Reply:
Hmm this is an interesting feature. I don't think it helps the overloads specifically, but looking back at them right now, I think perhaps they can be written with better type functions to condense some of them.

Reply:
This could be really interesting for typed transactions though. You could get back a transaction wherein types are `const` therefore showing you things like actual addresses, etc.